portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,30 +23,51 @@
 import com.google.gson.Gson;
 import java.util.ArrayList;
 import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.common.collect.ImmutableList;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.Streams;
 
 /** Servlet that handles comment data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private static final String COMMENT_NAME = "comment";
   private static final String DISPLAY_NAME = "name";
   private static final String DEFAULT_VAL = "";
-  private List<Comment> comments = new ArrayList<Comment>();
 
    @Override
-   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-     response.setContentType("application/json;");
-     Gson gson = new Gson();
-     String json = gson.toJson(comments);
-     response.getWriter().println(json);
-   }
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    ImmutableList<Comment> comments = 
+        Streams.stream(results.asIterable())
+            .map(entity -> (new Comment((String) entity.getProperty("comment"), (String) entity.getProperty("name"))))
```

Comment:
nit: Nice job on the streams! For readability, let's separate this into a helper method that takes an Entity and returns a new Comment. Also, let's define "comment" and "name" as constants (maybe something like COMMENT_PROPERTY and NAME_PROPERTY?)